sh_scrapy/settings.py

```diff
@@ -96,13 +95,25 @@ def _load_addons(addons, settings, merged_settings, priority=0):
         skey = _get_component_base(settings, addon['type'])
         components = settings[skey]
         path = update_classpath(addon_path)
-        components[path] = addon['order']
+        if path not in components:
+            components[path] = addon['order']
         merged_settings.set(skey, components)
         merged_settings.setdict(addon['default_settings'], priority)
 
 
+def _merge_with_keeping_order(settings, updates):
+    for setting, value in updates.items():
+        if not isinstance(value, dict):
+            settings.set(setting, value, priority='cmdline')
+            continue
+        for path, order in value.items():
+            if path not in settings[setting]:
```

Comment:
another place where path can be different from settings[settings] key because update_classpath is not applied after get_project_settings()